content/blog/2022-05-17-object-oriented-features/index.md

```diff
@@ -0,0 +1,210 @@
++++
+title = "Xi++: Object Oriented Features for Xi"
+[[extra.authors]]
+name = "Orko Sinha"
+link = ""
+[[extra.authors]]
+name = "Michael Maitland"
+link = "https://michaelmaitland.com"
++++
+Building a compiler for Xi is the main project in [CS 4120](https://www.cs.cornell.edu/courses/cs4120/) at Cornell. The final assignment, now discontinued, is to extend the compiler to support feaures of the [Xi++ Language Specification](https://www.cs.cornell.edu/courses/cs4120/2019sp/project/oolang.pdf) such as object-oriented programing. Our project is an implementation of these features on Michael's compiler created for CS 4120. Our project is [here](https://github.com/Yasgur99/xic/tree/oo).
+
+# Lexing
+For lexing, there are 5 new syntax elements that we need to handle, listed below:
```

Comment:
I think the Lexing and Parsing sections would be a lot more exciting if you were to just show a short snippet of code that uses this extension. What does the Xi++ code actually look like?

I also recommend combining Lexing and Parsing. The lexing stuff is so straightforward that it barely deserves mentioning. Better yet if you just skip over the parsing stuff as quickly as possible… knowing the syntax is useful, but I'm not sure there's much to learn from hearing about the parser itself.